include/translatable-object.php

```diff
@@ -246,53 +246,110 @@ public function delete_language( $id ) {
 	 *
-	 * @since 1.2
+	 * @since 3.8
 	 *
-	 * @param int    $id       Object ID.
-	 * @param string $taxonomy Polylang taxonomy depending if we are looking for a post (or term, or else) language.
-	 * @return WP_Term|false The term associated to the object in the requested taxonomy if it exists, `false` otherwise.
+	 * @param int[]  $object_ids Array of object IDs.
+	 * @param string $taxonomy   Taxonomy name.
+	 * @return array<int,WP_Term|null> Array of terms with object ID as key.
 	 */
-	public function get_object_term( $id, $taxonomy ) {
-		$id = $this->sanitize_int_id( $id );
+	protected function get_object_terms( array $object_ids, string $taxonomy ) {
+		$object_ids = $this->sanitize_int_ids_list( $object_ids );
+		if ( empty( $object_ids ) ) {
+			return array();
+		}
 
-		if ( empty( $id ) ) {
-			return false;
+		$this->update_object_term_cache( $object_ids );
+
+		$cached_values = wp_cache_get_multiple( $object_ids, "{$taxonomy}_relationships" );
+
+		// Flatten the array to prime the terms cache.
+		$all_term_ids = array();
+		foreach ( $cached_values as $term_ids ) {
+			$all_term_ids = array_merge( $all_term_ids, $term_ids );
+		}
+		_prime_term_caches( $all_term_ids, false );
+
+		$terms = array();
+		foreach ( $cached_values as $object_id => $term_ids ) {
+			if ( ! empty( $term_ids ) ) {
+				$term_id             = reset( $term_ids ); // There is only one term for language or translation groups.
+				$term                = get_term( $term_id );
+				$terms[ $object_id ] = $term instanceof WP_Term ? $term : null;
+			}
 		}
 
-		$term = get_object_term_cache( $id, $taxonomy );
+		return $terms;
+	}
+
+	/**
+	 * Caches all object-relationship terms.
+	 *
+	 * @since 3.8
+	 *
+	 * @param int[] $object_ids Array of object IDs to retrieve terms for.
+	 *
+	 * @return void
+	 */
+	private function update_object_term_cache( array $object_ids ): void {
+		$non_cached_ids = array();
+		foreach ( $this->tax_to_cache as $taxonomy ) {
+			$non_cached_ids = array_merge( $non_cached_ids, _get_non_cached_ids( $object_ids, "{$taxonomy}_relationships" ) );
+		}
 
-		if ( is_array( $term ) ) {
-			return ! empty( $term ) ? reset( $term ) : false;
+		if ( empty( $non_cached_ids ) ) {
+			return;
 		}
 
-		// Query terms.
-		$terms        = array();
-		$term         = false;
-		$object_terms = wp_get_object_terms( $id, $this->tax_to_cache, array( 'update_term_meta_cache' => false ) );
+		$terms = wp_get_object_terms(
+			array_unique( $non_cached_ids ),
+			$this->tax_to_cache,
+			array(
+				'fields'                 => 'all_with_object_id',
+				'update_term_meta_cache' => false,
+			)
+		);
 
-		if ( is_array( $object_terms ) ) {
-			foreach ( $object_terms as $t ) {
-				$terms[ $t->taxonomy ] = $t;
-				if ( $t->taxonomy === $taxonomy ) {
-					$term = $t;
-				}
+		if ( ! is_array( $terms ) ) {
+			return;
+		}
+
+		$object_terms = array();
+		foreach ( $terms as $term ) {
+			if ( ! isset( $term->object_id ) ) {
+				continue;
```

Comment:
I guess this is coming from PHPStan. However, we know that this property is always set. Is there a doc alternative to avoid useless operations?

Reply:
Done in 458fbb4c8f4da8e557ebf7e28a23c02d8bca7c90